app/controllers/dor_controller.rb

```diff
@@ -2,17 +2,21 @@
 
 # Main controller of application
 class DorController < ApplicationController
+  include Dry::Monads[:result]
+
+  class CocinaModelBuildError < StandardError; end
```

Comment:
i wasn't sure if wrapping the build error was overkill, but it made it easier to keep the building in a private helper method and the error handling in the `DorController#reindex` method, while still being sure that only a build error was being rescued (as opposed to rescuing all of `StandardError` in `#reindex`).